python/ray/experimental/dag/class_node.py

```diff
@@ -1,26 +1,51 @@
 import ray
-from ray.experimental.dag.dag_node import DAGNode
+import ray.experimental.dag as ray_dag
 
 from typing import Any, Dict, List, Optional, Tuple
 
 
-class ClassNode(DAGNode):
+class ClassNode(ray_dag.DAGNode):
     """Represents an actor creation in a Ray task DAG."""
 
-    def __init__(self, cls: type, cls_args, cls_kwargs, cls_options):
+    def __init__(
+        self,
+        cls: type,
+        cls_args,
+        cls_kwargs,
+        cls_options,
+        kwargs_to_resolve=None,
+    ):
+        if ray_dag.INPUT in cls_args or ray_dag.INPUT in cls_kwargs.values():
+            raise ValueError(
+                "dag.INPUT cannot be used as ClassNode args. "
+                "Please bind to function or class method only."
```

Comment:
```suggestion
                "dag.INPUT cannot be used as ClassNode args because it isn't available at class construction time. "
                "Please bind to function or class method only."
```
users will inevitably be confused as to why :)